crates/ark/src/interface.rs

```diff
@@ -282,9 +282,14 @@ pub enum ConsoleResult {
 }
 
 impl RMain {
-    /// Starts the main R thread and initializes the `R_MAIN` singleton.
-    /// Doesn't return. Must be called only once.
-    pub fn start(
+    /// Sets up the main R thread and initializes the `R_MAIN` singleton. Must
+    /// be called only once. This is doing as much setup as possible before
+    /// starting the R REPL. Since the REPL does not return, it might be
+    /// launched in a background thread (which we do in integration tests). The
+    /// setup can still be done in your main thread so that panics may propagate
```

Comment:
"so that panics" - do you mean panics during setup? or panics in general?

Reply:
during setup, I'll specify